flocx_market/db/sqlalchemy/api.py

```diff
@@ -163,15 +163,26 @@ def contract_destroy(contract_id):
 
 # offer_contract_relationship
 def offer_contract_relationship_get(marketplace_offer_id=None,
-                                    contract_id=None):
+                                    contract_id=None, status=None):
+
+    if (contract_id is not None) and (marketplace_offer_id is None)\
+            and (status is None):
+        return get_session().query(models.OfferContractRelationship).filter(
+            models.OfferContractRelationship.contract_id
+            == contract_id)
+
+    elif (marketplace_offer_id is not None) and (contract_id is None)\
+            and (status is None):
+        return get_session().query(models.OfferContractRelationship).filter(
+            models.OfferContractRelationship.marketplace_offer_id
+            == marketplace_offer_id)
+
+    elif (status is not None) and (contract_id is None)\
+            and (marketplace_offer_id is None):
+        return get_session().query(models.OfferContractRelationship).filter(
+            models.OfferContractRelationship.status
+            == status)
```

Comment:
Instead of this, you can chain filters together like this: https://github.com/openstack/ironic/blob/master/ironic/db/sqlalchemy/api.py#L302

Reply:
I'm not entirely sure which one are you referring to in this method? Since all these filters are filtering by a single element and requiring the uuid, and in our case we won't know the UUID. Can you explain how did you want that done @tzumainn?

Reply:
See answer above; it requires separating out `get` from `get_all`

Reply:
I just pushed the code with the get and get_all, let me know if that's what you were thinking about @tzumainn